### DISCUSSION TITLE
Session operation has timed out

### DISCUSSION THREAD

**Discussion by AkshayDeeti123** [sshnet/SSH.NET#1349] [Q&A] [answered]
We are currently using 2016.1.0 version to make SSH connection to the Cisco switches and execute some commands. And never faced any issue. 

Due to security concern, we have upgraded it to 2020.0.2. But it is leading to issues. i.e. Out of 50 SSH connections made to the switch and execute a command, only 27 are successful. Rest 23 failed with below error "Session operation has timed out".

Any help please to use 2020.0.2 ? We are unable to use the latest stable version 2024.0.0 because we are limited to use .Net Framework 4.5.2 now.

**Comment by Rob-Hague** [comment, marked as answer]
The fix is in #1274 which is included in 2023.0.1

I think you have 2 options:

1. Pull the source for 2020.0.2 and try to merge in #1274 
2. Pull the source for 2023.0.1 (or higher) and build it for 4.5.2

**Comment by WojciechNagorski** [comment]
You're right version 2016.1.0 contains vulnerabilities, and should be upgraded. 
.NET Framework 4.5.2 also contains vulnerabilities because it is unsupported. https://devblogs.microsoft.com/dotnet/net-framework-4-5-2-4-6-4-6-1-will-reach-end-of-support-on-april-26-2022/

We're not going to add support for the unsupported target framework.

Since version 2020.0.2, we have fixed many bugs and improved stability, performance and security. 
I think you have more options:
1. Update your project to the newer version of .NET Framework and take the latest version.
2. Create a small service in modern .NET that will be responsible for communication with SSH and use it in your legacy project. This service can be a microservice or exe - I don't know your project, so it's hard to guess.
